Sources/Networking/NetworkRequestPerformer.swift

```diff
@@ -28,4 +28,13 @@ public protocol NetworkRequestPerformer {
     /// - Returns: Returns a publisher that can be subscribed to, that performs the given request with the given behaviors.
     @available(iOS 13.0, *)
     @discardableResult func send(_ request: any NetworkRequest, requestBehaviors: [RequestBehavior]) -> AnyPublisher<NetworkResponse, NetworkError>
+    
+    /// Performs the given request with the given behaviors returning a JSON-decoded object with async/await, or throwing an error if unsuccessful.
+    ///
+    /// - Parameters:
+    ///   - request: The request to perform.
+    ///   - requestBehaviors: The behaviors to apply to the given request.
+    ///   - decoder: The JSON decoder to use when decoding the data.
+    /// - Returns: A JSON-decoded object.
+    func send<ResponseType: Decodable>(_ request: any NetworkRequest, requestBehaviors: [RequestBehavior], decoder: JSONDecoder) async throws -> ResponseType
```

Comment:
I don't think that we want to require JSON decoding support in the async function here in `NetworkRequestPerformer`, because it makes it so that any conforming type now has to implement JSON decoding. The JSON support was separated out into the extension for that reason. Instead, I think we'd either wan to have a non-parsing async version of the `send` function here that the implementation above then uses (probably preferred), or not include this function here